### ISSUE TITLE
How to check the "mAP" value from the detect.py?

### ISSUE THREAD

**Issue by jaskiratsingh2000** [ultralytics/yolov5#5069]
Hi @glenn-jocher I just want to ask that I want to check the mAP value of the TFlite model from the detect.py Is there any way I can get that value by passing some kind of parser?

FYI, I don't want to use the val.py script since it doesn't have tflite model implementation for mAP. Can you please let me know?

**Comment by glenn-jocher** [comment]
@jaskiratsingh2000 val.py is used for metrics computation on a dataset yaml:
```
python val.py --data coco128.yaml --weights yolov5s.pt
```

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher I think you haven't understood my question I actually want to measure the mAP value of tf.py model hence val.py doesn't provide the code implementation and I don't know how to do that is why I said that since mAP is a subset of detect.py so how  can we do that

**Comment by glenn-jocher** [comment]
@jaskiratsingh2000 yes I understand the feature request. This would be a useful feature to have. I will add this to our TODO queue to task future dev work on this, but it won't be ready anytime soon since we have many existing tasks to complete first. 

The best solution is probably an extensible multi-backend inference engine that val.py and detect.py can both call, which we can keep adding backends to, i.e. cv2 DNN, CoreML, etc.

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher How long I could expect this feature to come up?

In the meantime can you just try explaining me in bit more detail that what exactly I need to do to find mAP using detect.py ?

**Comment by glenn-jocher** [comment]
@jaskiratsingh2000 there's no timeline for this. I'm sorry I can't help with code customizations.

**Comment by jaskiratsingh2000** [comment]
@glenn-jocher Okay no worries. Also can you please look into my another
issue as well which talks about format of pytorch model?

**Comment by glenn-jocher** [comment]
@jaskiratsingh2000 good news 😃! Your original issue may now be fixed ✅ in PR #5549. This PR consolidates all backends into a single YOLOv5 `DetectMultiBackend()` class. Supports following model formats:

- [x] PyTorch `*.pt`
- [x] TorchScript `*.torchscript.pt`
- [x] CoreML `*.mlmodel`
- [x] TensorFlow `saved_model`
- [x] TensorFlow `*.pb`
- [x] TensorFlow Lite `*.tflite`
- [ ] TensorFlow.js `web_model`
- [x] ONNX Runtime `*.onnx`
- [x] OpenCV DNN `*.onnx`


### Usage
```python
# Export
python export.py --weights yolov5s.pt --include tflite

# Inference
python detect.py --weights yolov5s.tflite
python val.py --weights yolov5s.tflite
```

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
